### PR TITLE
Changed base GPU image to devel to compile Horovod CUDA kernels

### PR DESCRIPTION
**Pull request by tgaddair**
## Why are these changes needed?

The existing Ray GPU images do not come with the CUDA compiler toolchain, which prevents users from compiling custom CUDA kernels during install for frameworks like Horovod (see https://github.com/horovod/horovod/issues/2815).

By using the devel image instead of runtime from NVIDIA, Horovod and other compiled frameworks can use `nvcc` to build their custom CUDA kernels at install time. 

## Related issue number

https://github.com/horovod/horovod/issues/2815

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by tgaddair** [comment]
cc @richardliaw

**Review by richardliaw — COMMENTED** [review]
BTW, 

nvidia/cuda:11.0-cudnn8-runtime-ubuntu18.04: 1.66 GB

nvidia/cuda:11.0-cudnn8-devel-ubuntu18.04: 2.99 GB

I think this is OK, but @ijrsvt please take a look.

File: build-docker.sh
Comment (ijrsvt) [inline build-docker.sh:20]: Does [Tensorflow](https://www.tensorflow.org/install/gpu) work with Cuda 11.1?

**Comment by ijrsvt** [comment]
cc @ericl  @DmitriGekhtman 

Is the 1.5 GB size increase for `-gpu` images okay?

File: build-docker.sh
Comment (tgaddair) [reply build-docker.sh]: On Horovod, we've been successfully running TF built with CUDA 11.0 and PyTorch built with CUDA 11.1 using CUDA 11.2 image in our CI process: https://github.com/horovod/horovod/blob/master/docker-compose.test.yml#L200

Comment (tgaddair) [reply build-docker.sh]: This should probably be tested thoroughly, though, so to unblock I will change back to 11.0.

**Comment by richardliaw** [comment]
BTW, I think we should approve. This is because:

1. The ML image already contains a lot of bloat (torch AND tf). If users want a slim image they should build their own.
2. Making horovod easy to run on Ray Docker / K8s clusters is quite important, as we will be investing time into supporting horovod in the foreseeable future.

**Comment by ericl** [comment]
+1, I think the overhead is acceptable given we maintain the other image as
well.

File: build-docker.sh
Comment (ijrsvt) [inline build-docker.sh:20]:
```suggestion
    BASE_IMAGE="nvidia/cuda:11.0-cudnn8-devel-ubuntu18.04"
```

File: ci/travis/build-docker-images.py
Comment (ijrsvt) [inline ci/travis/build-docker-images.py:106]:
```suggestion
                    "nvidia/cuda:11.0-cudnn8-devel-ubuntu18.04"
```

File: docker/base-deps/Dockerfile
Comment (ijrsvt) [inline docker/base-deps/Dockerfile:3]:
```suggestion
# The GPU option is nvidia/cuda:11.0-cudnn8-devel-ubuntu18.04
```

**Review by ijrsvt — COMMENTED** [review]
This should be good (pending the change from `11.0.0` -> `11.0`) and ensuring that docker builds correctly :)

**Comment by tgaddair** [comment]
> This should be good (pending the change from `11.0.0` -> `11.0`) and ensuring that docker builds correctly :)

Thanks @ijrsvt, applied the changes.

**Review by richardliaw — APPROVED** [review]
Green! will merge this

**Review by richardliaw — APPROVED** [review]

**Pull request merged**
